runtime/common/src/assigned_slots/mod.rs

```diff
@@ -148,13 +178,41 @@ pub mod pallet {
 	#[pallet::getter(fn active_temporary_slot_count)]
 	pub type ActiveTemporarySlotCount<T: Config> = StorageValue<_, u32, ValueQuery>;
 
+	/// Assigned max temporary slots storage.
+	#[pallet::storage]
+	pub type MaxTemporarySlots<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+	/// Assigned max permanent slots storage.
+	#[pallet::storage]
+	pub type MaxPermanentSlots<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+	#[pallet::genesis_config]
+	#[derive(frame_support::DefaultNoBound)]
+	pub struct GenesisConfig<T: Config> {
+		pub max_temporary_slots: u32,
+		pub max_permanent_slots: u32,
+		pub _config: PhantomData<T>,
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+		fn build(&self) {
+			<MaxPermanentSlots<T>>::put(&self.max_permanent_slots);
+			<MaxTemporarySlots<T>>::put(&self.max_temporary_slots);
+		}
+	}
+
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		/// A para was assigned a permanent parachain slot
 		PermanentSlotAssigned(ParaId),
 		/// A para was assigned a temporary parachain slot
 		TemporarySlotAssigned(ParaId),
+		/// A maximum number of permanent slots has been assigned
+		MaxPermanentSlotsAssigned { slots: u32 },
+		/// A maximum number of temporary slots has been assigned
+		MaxTemporarySlotsAssigned { slots: u32 },
```

Comment:
I think this terminology is better suited here? 
```suggestion
		/// A maximum number of permanent slots has been changed
		MaxPermanentSlotsChanged { slots: u32 },
		/// A maximum number of temporary slots has been changed
		MaxTemporarySlotsChanged { slots: u32 },
```

Reply:
Agree, makes more sense